mapie/time_series_regression.py

```diff
@@ -0,0 +1,325 @@
+from __future__ import annotations
+
+from typing import Iterable, Optional, Tuple, Union, cast
+
+import numpy as np
+import numpy.ma as ma
+from sklearn.base import RegressorMixin
+from sklearn.model_selection import BaseCrossValidator
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+from .aggregation_functions import aggregate_all
+from .regression import MapieRegressor
+from ._typing import ArrayLike, NDArray
+from .utils import (
+    check_alpha,
+    check_alpha_and_n_samples,
+    masked_quantile,
+)
+
+
+class MapieTimeSeriesRegressor(MapieRegressor):
+    """
+    Prediction interval with out-of-fold residuals for time series.
+
+    This class implements the EnbPI strategy and some variations
+    for estimating prediction intervals on single-output time series.
+
+    Actually, EnbPI only corresponds to ``MapieTimeSeriesRegressor`` if the
+    ``cv`` argument if of type ``BlockBootstrap``.
+
+    References
+    ----------
+    Chen Xu, and Yao Xie.
+    [6] "Conformal prediction for dynamic time-series."
+    https://arxiv.org/abs/2010.09107
+    """
+
+    def __init__(
+        self,
+        estimator: Optional[RegressorMixin] = None,
+        method: str = "plus",
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None,
+        n_jobs: Optional[int] = None,
+        agg_function: Optional[str] = "mean",
+        verbose: int = 0,
+    ) -> None:
+        super().__init__(estimator, method, cv, n_jobs, agg_function, verbose)
+        self.cv_need_agg_function.append("BlockBootstrap")
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieTimeSeriesRegressor:
+        """
```

Comment:
Add a docstring explaining what is done in the `fit` method with respect to the one from `MapieRegressor` ?

Reply:
Done